app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/gainorlossafterreliefs/GainOrLossAfterReliefsController.scala

```diff
@@ -65,18 +69,25 @@ class GainOrLossAfterReliefsController @Inject() (
   def enterGainOrLossAfterReliefs(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
       withFillingOutReturnAndAnswers { (fillingOutReturn, draftReturn, answer) =>
-        Ok(
-          gainOrLossAfterReliefsPage(
-            answer.fold(gainOrLossAfterReliefsForm)(value => gainOrLossAfterReliefsForm.fill(value.inPounds())),
-            answer.fold(controllers.returns.routes.TaskListController.taskList())(_ =>
-              routes.GainOrLossAfterReliefsController.checkYourAnswers()
-            ),
-            fillingOutReturn.subscribedDetails.isATrust,
-            draftReturn.representativeType(),
-            draftReturn.triageAnswers().isLeft,
-            fillingOutReturn.isAmendReturn
+        glarCalculatorEligibilityUtil
+          .isEligibleForFurtherReturnOrAmendCalculation(fillingOutReturn)
+          .fold(
+            _ => InternalServerError,
```

Comment:
you could log the error here using `LoggerOps` in the `util` package and then you can return `errorHandler.errorResult()` which does some extra stuff and shows an actual page with content